releasenotes/notes/upgrade-pulse-builder-and-rzx-builder-033ac8ad8ad2a192.yaml

```diff
@@ -0,0 +1,23 @@
+---
+upgrade:
+  - |
+    The behavior of the pulse builder when a :class:`.Schedule` is called
+    has been upgraded. Called schedules are internally converted into
+    :class:`.ScheduleBlock` representation and now reference mechanism is
+    always applied rather than appending the schedules wrapped by
+    the :class:`~qiskit.pulse.instructions.Call` instruction.
+    Note that the converted block doesn't necessary recover the original alignment context.
+    This is simply an ASAP aligned sequence of pulse instructions with absolute time intervals.
+    This is an upgrade of internal representation of called pulse programs and thus no API changes.
+    However the :class:`~qiskit.pulse.instructions.Call` instruction and :class:`.Schedule`
+    no longer appear in the builder's pulse program.
+    This change guarantees the generated schedule blocks are always QPY compatible.
+  - |
+    :class:`~qiskit.transpiler.passes.RZXCalibrationBuilder`
+    and :class:`~qiskit.transpiler.passes.RZXCalibrationBuilderNoEcho` transpiler pass
+    have been upgraded to generate :class:`.ScheduleBlock`.
+    This change guarantees the transpiled circuits are always QPY compatible.
```

Comment:
For these release notes I'm wondering what the potential user change would need to be. We normally try to write these as something like: "we changed the behavior of X because of Y. If you were using X this means you will have do Z to adapt in the new release." For both of these I see X and Y, but not Z. I don't really want to bikeshed too much on release notes more, so this is more just a note for myself when we are doing the 0.23.0 release prep I can add the necessary details to these notes then.

Reply:
Thanks Matthew. The method is for almost internal use but it makes sense to have alternative path Z since the rescale method is public (I don't think this must be a public, apart from the unittest purpose). I added more description in [b17595e](https://github.com/Qiskit/qiskit-terra/pull/8949/commits/b17595e08f1de66ee39bcb240493385d2039270d)